src/runtime/stream.rs

```diff
@@ -127,12 +133,11 @@ impl AsyncStream {
         // If there are TLS options, wrap the inner stream with rustls.
         match options.tls_options {
             Some(cfg) => {
-                let name = DNSNameRef::try_from_ascii_str(&options.address.hostname)?;
```

Comment:
`connector.connect` [does this for us now, and it automatically converts the error to an io error](https://docs.rs/async-tls/0.6.0/src/async_tls/connector.rs.html#109). Because of that, I removed the `InvalidDnsName` error type as it is no longer being returned anywhere.